htlcswitch/switch.go

```diff
@@ -63,6 +64,52 @@ var (
 	ErrUnreadableFailureMessage = errors.New("unreadable failure message")
 )
 
+// forwardResolver implements the htlcinterceptor.ForwardResolver interface.
+// It is passed from the switch to external interceptors that are interested
+// in holding forwards and resolve them manually.
+type forwardResolver struct {
+	packet     *htlcPacket
+	htlcSwitch *Switch
+}
+
+// Resume resumes the default behavior as if the packet was not intercepted.
+func (h *forwardResolver) Resume() error {
+	pkt := *h.packet
+	pkt.intercepted = true
+	return h.htlcSwitch.route(&pkt)
```

Comment:
Yes, this is the main event loop. The only change is the `intercepted` flag that is sent to true to prevent from the switch to intercept the packet again.